src/cargo/ops/cargo_compile.rs

```diff
@@ -284,12 +288,76 @@ pub fn compile_ws<'a>(
         unit_graph::emit_serialized_unit_graph(&bcx.roots, &bcx.unit_graph)?;
         return Ok(Compilation::new(&bcx)?);
     }
-
     let _p = profile::start("compiling");
     let cx = Context::new(&bcx)?;
     cx.compile(exec)
 }
 
+pub fn print<'a>(
+    ws: &Workspace<'a>,
+    options: &CompileOptions,
+    print_opt_value: &str,
+) -> CargoResult<()> {
+    let CompileOptions {
+        ref build_config,
+        ref target_rustc_args,
+        ..
+    } = *options;
+    let config = ws.config();
+    let rustc = config.load_global_rustc(Some(ws))?;
+    let mut kinds = build_config.requested_kinds.clone();
+    if kinds.is_empty() {
+        kinds.push(CompileKind::Host);
+    }
+    let mut print_empty_line = false;
+    for kind in kinds {
+        let rustflags = env_args(
+            config,
+            &build_config.requested_kinds,
+            &rustc.host,
+            None,
+            kind,
+            "RUSTFLAGS",
+        )?;
+        let mut process = rustc.process();
+        process
+            .arg("-")
+            .arg("--crate-name")
+            .arg("___")
+            .args(&rustflags);
+        if let Some(args) = target_rustc_args {
+            process.args(args);
+        }
+        if let CompileKind::Target(t) = kind {
+            process.arg("--target").arg(t.short_name());
+        }
+        process
+            .arg("--print")
+            .arg(print_opt_value)
+            .env_remove("RUSTC_LOG");
```

Comment:
This is ok to remove. If it should happen it should happen in `rustc.process()`, but otherwise I don't think there's any reason to single it out here.